scripts/run-live-build.sh

```diff
@@ -129,36 +129,64 @@ while ! curl --output /dev/null --silent --head --fail \
 done
 set -o errexit
 
+pkg_mirror_secondary=''
 if [[ -n "$DELPHIX_PACKAGE_MIRROR_SECONDARY" ]]; then
-	sed "s|@@URL@@|$DELPHIX_PACKAGE_MIRROR_SECONDARY|" \
-		<config/archives/delphix-secondary-mirror.list.in \
-		>config/archives/delphix-secondary-mirror.list
+	pkg_mirror_secondary="$DELPHIX_PACKAGE_MIRROR_SECONDARY"
+else
+	#
+	# If no secondary package mirror is provided, then pull in the latest
+	# mirror dataset for the build. If no latest dataset is found, then fail.
+	#
+	source_url="http://linux-package-mirror.delphix.com/$UPSTREAM_PRODUCT_BRANCH/latest/"
+	pkg_mirror_secondary="$(curl -Ls -o /dev/null -w '%{url_effective}' $source_url)ppas"
+	if [[ "$pkg_mirror_secondary" == "${source_url}ppas" ]]; then
```

Comment:
Under what circumstances may this condition return true? e.g. what guarantees do we have about what the above `curl` command will return; when will it return the "correct" value, and when may it return an "incorrect" value? Is the only way to determine "correct vs. incorrect" by doing a comparison like this, with the original "source url"?

Reply:
Since I am using the -s (--silent) flag (since I just want the resulting URL) this returns `true` if the server responds to the HEAD request. Looking into this more, I think I should add the `--show-error` option as well so that the curl will fail if the URL doesn't exist. Then I won't have to do the comparison, and can just check the error message of the curl call. I will update to do that.